### DISCUSSION TITLE
Using Synology DDNS with AIO

### DISCUSSION THREAD

**Discussion by hunterwinterton** [nextcloud/all-in-one#565] [Questions] [answered]
I am wondering if anyone as successfully setup AIO to work with a DDNS provided by Synology. I have created a reverse proxy, nextcloud.synologyddns.me, and configured it to redirect to 11000, and also port forwarded my router, but when I try to create a new AIO instance with this DDNS, I get an error, "DNS config is not set or domain is not in a valid format!". Has anyone else had this problem and knows a solution?

**Comment by szaimen** [comment]
can you post the docker command that you used?

**Reply by hunterwinterton** [reply]
sudo docker run -it \
--name nextcloud-aio-mastercontainer \
--restart always \
-p 8080:8080 \
-e APACHE_PORT=11000 \
-e NEXTCLOUD_DATADIR="/host_mnt/volume1/docker/nextcloud/data" \
--volume nextcloud_aio_mastercontainer:/mnt/docker-aio-config \
--volume /var/run/docker.sock:/var/run/docker.sock:ro \
nextcloud/all-in-one:latest

**Reply by szaimen** [reply]
that looks good

**Reply by szaimen** [reply]
you now only need to configure your reverse proxy correctly to foward connections to your the correct container...

**Reply by szaimen** [reply]
which reverse proxy are you using?

**Reply by hunterwinterton** [reply]
<img width="583" alt="Screen Shot 2022-05-04 at 11 13 34 AM" src="https://user-images.githubusercontent.com/68767432/166742786-5aa725c2-f1fc-462d-9004-8d380a75b242.png">
Synologys built in reverse proxy

**Reply by szaimen** [reply]
does it handle the ssl certificates automatically?

**Reply by szaimen** [reply]
Also, does the synology domain have a valid ipv4 record?

**Reply by hunterwinterton** [reply]
Yes, I have a wildcard certificate so it has an ssl certificate. 
I don't know what a ipv4 record, unless its what public ip my DDNS points too. In that case, yes, it is connected to my public ip and all of my other domains work fine.

**Reply by szaimen** [reply]
what does the command `dig your.domain.com` return?

**Reply by szaimen** [reply]
basically it seem to fail here which means that the gethostbyname command does not return a vaild ip-address:
![image](https://user-images.githubusercontent.com/42591237/166746953-ed590c6e-a899-447e-9ca1-cb0e83aa9da5.png)

**Reply by hunterwinterton** [reply]
; <<>> DiG 9.10.6 <<>> nextcloud.name.synology.me
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 46504
;; flags: qr rd ra; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 4096
;; QUESTION SECTION:
;nextcloud.name.synology.me. IN A

;; ANSWER SECTION:
nextcloud.name.synology.me. 240 IN A	mypublicip

;; Query time: 1908 msec
;; SERVER: currentnetwork
;; WHEN: Wed May 04 11:39:27 MDT 2022
;; MSG SIZE  rcvd: 82

**Reply by szaimen** [reply]
hm... that looks good to me. I don't have any idea than why it fails at this step. This shoud<sup>TM</sup> work

**Reply by szaimen** [reply]
I suppose you've entered nextcloud.name.synology.me into the input field?
(Without http or https)

**Reply by hunterwinterton** [reply]
Yeah, I only put the domain name.

**Reply by szaimen** [reply]
Hm...

**Reply by hunterwinterton** [reply]
Is there a way to setup the container so I dont need to put in my domain name?

**Reply by szaimen** [reply]
unfortunately no. The question is why the function does not work inside the container... 
can you run the dig command from inside the container?
you can get into the container by runing `sudo docker exec -it nextcloud-aio-mastercontainer bash`
and then run the dig command from there...

**Reply by hunterwinterton** [reply]
I get command not found.

**Reply by szaimen** [reply]
Inside the container:
apt update && apt install dnsutils
And then please try again the dig command

**Reply by hunterwinterton** [reply]
apt update failed because it couldn't resolve any domains.. something in the network or my DNS must be messed up.

**Reply by szaimen** [reply]
Yes, that is what I thought. This explains also why the domaincheck fails...

**Reply by hunterwinterton** [reply]
DNS is working on the rest of my NAS, why would it not work on this docker container? Is it something with the bridge network?

**Reply by szaimen** [reply]
This goes deep into docker networking. Not sure if I am able to help here... I think the dns server can be set in some config file for the docker daemon...

**Reply by szaimen** [reply]
The Container uses whatever dns server is configured in this docker daemon config file, but it defaults to 8.8.8.8 usually, afaik.

**Reply by hunterwinterton** [reply]
I was finally able to fix the problem, I added a --dns="pihole ip", which made it so I had dns resolution in the container!

**Reply by hunterwinterton** [reply]
Now I have another issue after installing the different containers, my nextcloud-aio-nextcloud won't start because "Bond mount failed: '/host_mnt/volume1/docker/nextcloud/data' does not exist" Is this the correct way to mount a folder on my Synology to the nextcloud? I know on other containers it is different, I usually do "/volume1/docker/nextcloud/data:/var/data/"

**Reply by szaimen** [reply]
Great that you found a solution to the dns problem!
And, yes /volume1/... would be correct.
Is /volumeX/ always the default mountpoints of volumes on synology? If yes, we can probably allow that as well.

**Reply by szaimen** [reply]
I can create a PR that allows these paths too if you confirm.

**Reply by szaimen** [reply]
Okay, googled a bit and this seems to be indeed the case

**Reply by szaimen** [reply]
I'll create a pr

**Reply by hunterwinterton** [reply]
So would I just put: /volume1/docker/nextcloud/data, or do I need to bind it to a folder the container has? Like /volume1/:nextcloud/data

**Reply by szaimen** [reply]
/volume1/docker/nextcloud/data looks good but is currently not supported... you unfortunatley will need to wait for the next update...

**Reply by szaimen** [reply]
I'll also make the dns problem better debuggable in the future which also will be part of the next release

**Reply by hunterwinterton** [reply]
Any idea when the next update will be?

**Reply by szaimen** [reply]
I'll keep you updated on this. Depending on the things that need to go in probably next week.

**Reply by szaimen** [reply]
Hi, just released 1.2.0 Beta. You can test this already by using the beta channel.
See https://github.com/nextcloud/all-in-one#how-to-switch-the-channel

**Reply by hunterwinterton** [reply]
I just updated, and I made sure to remove the old images, but I still got the same error: You've set NEXTCLOUD_DATADIR but not to an allowed value.
The string must start with '/mnt/', '/media/' or '/host_mnt/'. E.g. '/mnt/ncdata'

sudo docker run -it \
--name nextcloud-aio-mastercontainer \
--dns="192.168.2.2" \
--restart always \
-p 8080:8080 \
-e APACHE_PORT=11000 \
-e NEXTCLOUD_DATADIR="/volume1/docker/nextcloud/data" \
--volume nextcloud_aio_mastercontainer:/mnt/docker-aio-config \
--volume /var/run/docker.sock:/var/run/docker.sock:ro \
nextcloud/all-in-one:latest

**Reply by szaimen** [reply]
You need to use the beta channel since 1.2.0 is not released yet to the latest channel. See https://github.com/nextcloud/all-in-one#how-to-switch-the-channel

**Reply by hunterwinterton** [reply]
Oh, I just looked at the docker page and it said it was just updated so that’s why I thought it wasn’t beta anymore. I will try the beta

**Reply by hunterwinterton** [reply]
Now I get an error in the container logs: [Mon May 09 18:34:36.239478 2022] [access_compat:error] [pid 111] [client 172.17.0.1:60468] AH01797: client denied by server configuration: proxy:http://localhost:8000/ | stdout

And this error in the browser
Forbidden

You don't have permission to access this resource.

**Reply by szaimen** [reply]
Alright. Can you try to access the interface from a different PC with a different ip-address?

**Reply by hunterwinterton** [reply]
I tried 3 different devices and same issue.

**Reply by szaimen** [reply]
okay, then feel free to switch back to the latest channel. I guess this is something we should address before releasing to latest...

**Reply by szaimen** [reply]
This should be fixed in the beta channel in the meantime.
So after a `sudo docker pull nextcloud/all-in-one:beta` it should allow you to access the aio interface again!
And thanks for testing, btw!

**Reply by hunterwinterton** [reply]
Now apache and nextcloud containers won't start after installing.
nc: getaddrinfo for host "nextcloud-aio-nextcloud" port 9000: No address associated with hostname
Waiting for Nextcloud to start...
Also, how do I clear the nextcloud files so I can start from scratch when starting the docker?

**Reply by szaimen** [reply]
Hm...

You can start from scratch by stopping all the aio containers, including the mastercontainer, then run `sudo docker container prune` and then `sudo docker volume prune`. Afterwards try again with the startup command.

**Reply by hunterwinterton** [reply]
So I just pruned the containers and volumes and reinstalled the container, and I am getting the same error.

**Reply by hunterwinterton** [reply]
In the nextcloud container I am getting:
Configuring Redis as session handler
touch: /mnt/ncdata/this-is-a-test-file: Permission denied
The www-data user doesn't seem to have access rights in /mnt/ncdata.
Did you maybe change the datadir and did forget to apply the correct permissions?

Does this have something to do with me mounting the data folder to my Synology?

**Reply by szaimen** [reply]
> So I just pruned the containers and volumes and reinstalled the container, and I am getting the same error.

Sorry, currently no idea why this happens to you or how to workaround

**Reply by szaimen** [reply]
> In the nextcloud container I am getting: Configuring Redis as session handler touch: /mnt/ncdata/this-is-a-test-file: Permission denied The www-data user doesn't seem to have access rights in /mnt/ncdata. Did you maybe change the datadir and did forget to apply the correct permissions?
> 
> Does this have something to do with me mounting the data folder to my Synology?

Yes, see 
![Screenshot_20220510-021119_Brave](https://user-images.githubusercontent.com/42591237/167518119-b0861a94-9a67-4869-becf-6896dd09ea2d.jpg)

**Reply by szaimen** [reply]
> > So I just pruned the containers and volumes and reinstalled the container, and I am getting the same error.
> 
> Sorry, currently no idea why this happens to you or how to workaround

Okay, this probably happens then because the nextcloud container is not able to start because of missing permissions.

**Reply by hunterwinterton** [reply]
I did the chown and still get the error after restarting :/

**Reply by szaimen** [reply]
Can you send the output of `sudo docker inspect nextcloud-aio-nextcloud` please? Of course feel free to mask the sensitive infos

**Reply by hunterwinterton** [reply]
[
    {
        "Id": "3d567c78f091be5e7e05e076b965a6765e959aaf1ee2384c9b64bf919e1074e2",
        "Created": "2022-05-10T00:19:58.657967576Z",
        "Path": "/start.sh",
        "Args": [
            "/usr/bin/supervisord",
            "-c",
            "/supervisord.conf"
        ],
        "State": {
            "Status": "restarting",
            "Running": true,
            "Paused": false,
            "Restarting": true,
            "OOMKilled": false,
            "Dead": false,
            "Pid": 0,
            "ExitCode": 1,
            "Error": "",
            "StartedAt": "2022-05-10T01:45:45.939491251Z",
            "FinishedAt": "2022-05-10T01:45:46.496425153Z",
            "StartedTs": 1652147145,
            "FinishedTs": 1652147146
        },
        "Image": "sha256:1fddb72634a37649becacbf89e2da8b8cc01358f59d771e56dbe18d32ce4e65b",
        "ResolvConfPath": "/volume1/@docker/containers/3d567c78f091be5e7e05e076b965a6765e959aaf1ee2384c9b64bf919e1074e2/resolv.conf",
        "HostnamePath": "/volume1/@docker/containers/3d567c78f091be5e7e05e076b965a6765e959aaf1ee2384c9b64bf919e1074e2/hostname",
        "HostsPath": "/volume1/@docker/containers/3d567c78f091be5e7e05e076b965a6765e959aaf1ee2384c9b64bf919e1074e2/hosts",
        "LogPath": "/volume1/@docker/containers/3d567c78f091be5e7e05e076b965a6765e959aaf1ee2384c9b64bf919e1074e2/log.db",
        "Name": "/nextcloud-aio-nextcloud",
        "RestartCount": 89,
        "Driver": "btrfs",
        "Platform": "linux",
        "MountLabel": "",
        "ProcessLabel": "",
        "AppArmorProfile": "docker-default",
        "ExecIDs": null,
        "HostConfig": {
            "Binds": [
                "nextcloud_aio_nextcloud:/var/www/html:rw",
                "/volume1/docker/nextcloud/data:/mnt/ncdata:rw"
            ],
            "ContainerIDFile": "",
            "LogConfig": {
                "Type": "db",
                "Config": {}
            },
            "NetworkMode": "default",
            "PortBindings": null,
            "RestartPolicy": {
                "Name": "unless-stopped",
                "MaximumRetryCount": 0
            },
            "AutoRemove": false,
            "VolumeDriver": "",
            "VolumesFrom": null,
            "CapAdd": null,
            "CapDrop": null,
            "CgroupnsMode": "host",
            "Dns": null,
            "DnsOptions": null,
            "DnsSearch": null,
            "ExtraHosts": null,
            "GroupAdd": null,
            "IpcMode": "private",
            "Cgroup": "",
            "Links": null,
            "OomScoreAdj": 0,
            "PidMode": "",
            "Privileged": false,
            "PublishAllPorts": false,
            "ReadonlyRootfs": false,
            "SecurityOpt": null,
            "UTSMode": "",
            "UsernsMode": "",
            "ShmSize": 67108864,
            "Runtime": "runc",
            "Env": [
                "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
                "PHPIZE_DEPS=autoconf \t\tdpkg-dev dpkg \t\tfile \t\tg++ \t\tgcc \t\tlibc-dev \t\tmake \t\tpkgconf \t\tre2c",
                "PHP_INI_DIR=/usr/local/etc/php",
                "PHP_CFLAGS=-fstack-protector-strong -fpic -fpie -O2 -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64",
                "PHP_CPPFLAGS=-fstack-protector-strong -fpic -fpie -O2 -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64",
                "PHP_LDFLAGS=-Wl,-O1 -pie",
                "GPG_KEYS=1729F83938DA44E27BA0F4D3DBDB397470D12172 BFDDD28642824F8118EF77909B67A5C12229118F",
                "PHP_VERSION=8.0.18",
                "PHP_URL=https://www.php.net/distributions/php-8.0.18.tar.xz",
                "PHP_ASC_URL=https://www.php.net/distributions/php-8.0.18.tar.xz.asc",
                "PHP_SHA256=db161652cacae4b31c347fbf2e17b80656473cb365f2bb3460c4552f5647e2e7",
                "PHP_MEMORY_LIMIT=512M",
                "PHP_UPLOAD_LIMIT=10G",
                "NEXTCLOUD_VERSION=23.0.4",
                "NEXTCLOUD_UPDATE=1",
                "POSTGRES_HOST=nextcloud-aio-database",
                "POSTGRES_PASSWORD=e4dff8ab337e68b649e65e25f72088149534fb3f578f6e38",
                "POSTGRES_DB=nextcloud_database",
                "POSTGRES_USER=nextcloud",
                "REDIS_HOST=nextcloud-aio-redis",
                "REDIS_HOST_PASSWORD=9481ad8cc2fb36d80e38723ceba86380eb1561ad03a0ce6a",
                "AIO_TOKEN=edc8d3e07b84d6dd098c50d012e082038cb3b4f6bc7ff07b",
                "NC_DOMAIN=redacted",
                "ADMIN_USER=admin",
                "ADMIN_PASSWORD=c7bd86bfb800fa4ca5ae81a99476b9524f54e02346f92ea0",
                "NEXTCLOUD_DATA_DIR=/mnt/ncdata",
                "OVERWRITEHOST=redacted",
                "OVERWRITEPROTOCOL=https",
                "TURN_SECRET=9aff9eeaed7d735f6926e57a52894561525ac3071beec75e",
                "SIGNALING_SECRET=bc0e42b7e173a0f33c44d5715d32dc958b33628886e9a779",
                "AIO_URL=192.168.1.125:8080",
                "NEXTCLOUD_MOUNT=",
                "CLAMAV_ENABLED=",
                "CLAMAV_HOST=nextcloud-aio-clamav",
                "ONLYOFFICE_ENABLED=",
                "COLLABORA_ENABLED=yes",
                "COLLABORA_HOST=nextcloud-aio-collabora",
                "TALK_ENABLED=yes",
                "ONLYOFFICE_HOST=nextcloud-aio-onlyoffice",
                "DAILY_BACKUP_RUNNING="
            ],
            "ConsoleSize": [
                0,
                0
            ],
            "Isolation": "",
            "CpuShares": 0,
            "Memory": 0,
            "NanoCpus": 0,
            "CgroupParent": "",
            "BlkioWeight": 0,
            "BlkioWeightDevice": null,
            "BlkioDeviceReadBps": null,
            "BlkioDeviceWriteBps": null,
            "BlkioDeviceReadIOps": null,
            "BlkioDeviceWriteIOps": null,
            "CpuPeriod": 0,
            "CpuQuota": 0,
            "CpuRealtimePeriod": 0,
            "CpuRealtimeRuntime": 0,
            "CpusetCpus": "",
            "CpusetMems": "",
            "Devices": null,
            "DeviceCgroupRules": null,
            "DeviceRequests": null,
            "KernelMemory": 0,
            "KernelMemoryTCP": 0,
            "MemoryReservation": 0,
            "MemorySwap": 0,
            "MemorySwappiness": null,
            "OomKillDisable": false,
            "PidsLimit": null,
            "Ulimits": null,
            "CpuCount": 0,
            "CpuPercent": 0,
            "IOMaximumIOps": 0,
            "IOMaximumBandwidth": 0,
            "MaskedPaths": [
                "/proc/asound",
                "/proc/acpi",
                "/proc/kcore",
                "/proc/keys",
                "/proc/latency_stats",
                "/proc/timer_list",
                "/proc/timer_stats",
                "/proc/sched_debug",
                "/proc/scsi",
                "/sys/firmware"
            ],
            "ReadonlyPaths": [
                "/proc/bus",
                "/proc/fs",
                "/proc/irq",
                "/proc/sys",
                "/proc/sysrq-trigger"
            ]
        },
        "GraphDriver": {
            "Data": null,
            "Name": "btrfs"
        },
        "Mounts": [
            {
                "Type": "volume",
                "Name": "nextcloud_aio_nextcloud",
                "Source": "/volume1/@docker/volumes/nextcloud_aio_nextcloud/_data",
                "Destination": "/var/www/html",
                "Driver": "local",
                "Mode": "rw",
                "RW": true,
                "Propagation": ""
            },
            {
                "Type": "bind",
                "Source": "/volume1/docker/nextcloud/data",
                "Destination": "/mnt/ncdata",
                "Mode": "rw",
                "RW": true,
                "Propagation": "rprivate"
            }
        ],
        "Config": {
            "Hostname": "3d567c78f091",
            "Domainname": "",
            "User": "www-data",
            "AttachStdin": false,
            "AttachStdout": false,
            "AttachStderr": false,
            "ExposedPorts": {
                "9000/tcp": {}
            },
            "Tty": false,
            "OpenStdin": false,
            "StdinOnce": false,
            "Env": [
                "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
                "PHPIZE_DEPS=autoconf \t\tdpkg-dev dpkg \t\tfile \t\tg++ \t\tgcc \t\tlibc-dev \t\tmake \t\tpkgconf \t\tre2c",
                "PHP_INI_DIR=/usr/local/etc/php",
                "PHP_CFLAGS=-fstack-protector-strong -fpic -fpie -O2 -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64",
                "PHP_CPPFLAGS=-fstack-protector-strong -fpic -fpie -O2 -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64",
                "PHP_LDFLAGS=-Wl,-O1 -pie",
                "GPG_KEYS=1729F83938DA44E27BA0F4D3DBDB397470D12172 BFDDD28642824F8118EF77909B67A5C12229118F",
                "PHP_VERSION=8.0.18",
                "PHP_URL=https://www.php.net/distributions/php-8.0.18.tar.xz",
                "PHP_ASC_URL=https://www.php.net/distributions/php-8.0.18.tar.xz.asc",
                "PHP_SHA256=db161652cacae4b31c347fbf2e17b80656473cb365f2bb3460c4552f5647e2e7",
                "PHP_MEMORY_LIMIT=512M",
                "PHP_UPLOAD_LIMIT=10G",
                "NEXTCLOUD_VERSION=23.0.4",
                "NEXTCLOUD_UPDATE=1",
                "POSTGRES_HOST=nextcloud-aio-database",
                "POSTGRES_PASSWORD=e4dff8ab337e68b649e65e25f72088149534fb3f578f6e38",
                "POSTGRES_DB=nextcloud_database",
                "POSTGRES_USER=nextcloud",
                "REDIS_HOST=nextcloud-aio-redis",
                "REDIS_HOST_PASSWORD=9481ad8cc2fb36d80e38723ceba86380eb1561ad03a0ce6a",
                "AIO_TOKEN=edc8d3e07b84d6dd098c50d012e082038cb3b4f6bc7ff07b",
                "NC_DOMAIN=redacted",
                "ADMIN_USER=admin",
                "ADMIN_PASSWORD=c7bd86bfb800fa4ca5ae81a99476b9524f54e02346f92ea0",
                "NEXTCLOUD_DATA_DIR=/mnt/ncdata",
                "OVERWRITEHOST=redacted",
                "OVERWRITEPROTOCOL=https",
                "TURN_SECRET=9aff9eeaed7d735f6926e57a52894561525ac3071beec75e",
                "SIGNALING_SECRET=bc0e42b7e173a0f33c44d5715d32dc958b33628886e9a779",
                "AIO_URL=192.168.1.125:8080",
                "NEXTCLOUD_MOUNT=",
                "CLAMAV_ENABLED=",
                "CLAMAV_HOST=nextcloud-aio-clamav",
                "ONLYOFFICE_ENABLED=",
                "COLLABORA_ENABLED=yes",
                "COLLABORA_HOST=nextcloud-aio-collabora",
                "TALK_ENABLED=yes",
                "ONLYOFFICE_HOST=nextcloud-aio-onlyoffice",
                "DAILY_BACKUP_RUNNING="
            ],
            "Cmd": [
                "/usr/bin/supervisord",
                "-c",
                "/supervisord.conf"
            ],
            "Image": "nextcloud/aio-nextcloud:beta",
            "Volumes": {
                "/mnt/ncdata": {},
                "/var/www/html": {}
            },
            "WorkingDir": "/var/www/html",
            "Entrypoint": [
                "/start.sh"
            ],
            "OnBuild": null,
            "Labels": {},
            "StopSignal": "SIGQUIT",
            "DDSM": false
        },
        "NetworkSettings": {
            "Bridge": "",
            "SandboxID": "7770ed49d6952aea7c724fbd524ff5e864f0ee71835ca9f3c3c9b5da4cd10d91",
            "HairpinMode": false,
            "LinkLocalIPv6Address": "",
            "LinkLocalIPv6PrefixLen": 0,
            "Ports": {},
            "SandboxKey": "/var/run/docker/netns/7770ed49d695",
            "SecondaryIPAddresses": null,
            "SecondaryIPv6Addresses": null,
            "EndpointID": "",
            "Gateway": "",
            "GlobalIPv6Address": "",
            "GlobalIPv6PrefixLen": 0,
            "IPAddress": "",
            "IPPrefixLen": 0,
            "IPv6Gateway": "",
            "MacAddress": "",
            "Networks": {
                "bridge": {
                    "IPAMConfig": null,
                    "Links": null,
                    "Aliases": null,
                    "NetworkID": "d70d64f68fda72551f64c21457ed13ba9950be73d52a921b95ef0ddfe9f5f5e3",
                    "EndpointID": "",
                    "Gateway": "",
                    "IPAddress": "",
                    "IPPrefixLen": 0,
                    "IPv6Gateway": "",
                    "GlobalIPv6Address": "",
                    "GlobalIPv6PrefixLen": 0,
                    "MacAddress": "",
                    "DriverOpts": null
                },
                "nextcloud-aio": {
                    "IPAMConfig": null,
                    "Links": null,
                    "Aliases": [
                        "3d567c78f091"
                    ],
                    "NetworkID": "748db4abae8d65f4004e809b5fec135789e80f7f472fdbeb26674163ce1fe6d9",
                    "EndpointID": "",
                    "Gateway": "",
                    "IPAddress": "",
                    "IPPrefixLen": 0,
                    "IPv6Gateway": "",
                    "GlobalIPv6Address": "",
                    "GlobalIPv6PrefixLen": 0,
                    "MacAddress": "",
                    "DriverOpts": null
                }
            }
        }
    }
]

**Reply by szaimen** [reply]
Thanks! 

So you have run `sudo chown -R 33:0 /volume1/docker/nextcloud/data` and it still does not work after a restart?

What is the output of `sudo ls -la /volume1/docker/nextcloud/data`?

**Reply by hunterwinterton** [reply]
total 0
dr-xr-xr-x+ 1                    33 root   26 May  3 20:47 .
drwxr-xr-x  1 adminname users  74 May  9 16:06 ..
dr-xr-xr-x+ 1                    33 root    6 Dec  6 22:02 Luminar
dr-xr-xr-x+ 1                    33 root  162 May  3 18:53 Photos

**Reply by szaimen** [reply]
I see. Write permissions are missing. Can you run `sudo chmod 770 -R /volume1/docker/nextcloud/data` and see if it fixes the issue?

**Reply by hunterwinterton** [reply]
Thanks so much for your help, that fixed it! One last question, when I have my Synology firewall turned on, the containers don't seem to start. I added the ports that the documentation asks for, like 8080, 11000, and 3478. Are there any other ports I need to add?

**Reply by szaimen** [reply]
Internally are indeed other ports needed but I don't think the firewall should hinder the containers from starting?
Which containers are affected?

**Reply by hunterwinterton** [reply]
Collabra, Redis, and Database

**Reply by szaimen** [reply]
Sorry no idea what the problem now is... what do the container logs say?

**Reply by hunterwinterton** [reply]
Database log says:
2022-05-10 12:52:07.322 UTC [1544] ERROR:  duplicate key value violates unique constraint "fs_storage_path_hash"
2022-05-10 12:52:07.322 UTC [1544] DETAIL:  Key (storage, path_hash)=(2, 34ff2c2a226d6a795816acabcd07d8f7) already exists.
2022-05-10 12:52:07.322 UTC [1544] STATEMENT:  INSERT INTO "oc_filecache" ("mimepart", "mimetype", "mtime", "size", "etag", "storage_mtime", "permissions", "name", "parent", "checksum", "path_hash", "path", "storage") VALUES($1, $2, $3, $4, $5, $6, $7, $8, $9, $10, $11, $12, $13)

**Reply by szaimen** [reply]
and the others?

**Reply by hunterwinterton** [reply]
Collabra: 
wsd-00001-00033 2022-05-11 00:12:43.555596 +0000 [ prisoner_poll ] TRC  ppoll start, timeoutMicroS: 5000000 size 1| net/Socket.cpp:339
frk-00034-00034 2022-05-11 00:12:43.642989 +0000 [ coolforkit ] INF  Initializing frk. Local time: Wed 2022-05-11 00:12:43+0000. Log level is [8].| common/Log.cpp:391
frk-00034-00034 2022-05-11 00:12:43.643021 +0000 [ coolforkit ] INF  Setting log-level to [trace] and delaying setting to configured [warning] until after Forkit initialization.| kit/ForKit.cpp:566
frk-00034-00034 2022-05-11 00:12:43.643043 +0000 [ coolforkit ] INF  RLIMIT_AS is unlimited after setting it to unlimited.| common/Seccomp.cpp:283
frk-00034-00034 2022-05-11 00:12:43.643063 +0000 [ coolforkit ] INF  RLIMIT_STACK is 8192000 bytes after setting it to 8192000 bytes.| common/Seccomp.cpp:283
frk-00034-00034 2022-05-11 00:12:43.643074 +0000 [ coolforkit ] INF  Ignored setting RLIMIT_FSIZE to unlimited.| common/Seccomp.cpp:289
frk-00034-00034 2022-05-11 00:12:43.643086 +0000 [ coolforkit ] INF  Ignored setting RLIMIT_NOFILE to unlimited.| common/Seccomp.cpp:289
frk-00034-00034 2022-05-11 00:12:43.643120 +0000 [ coolforkit ] DBG  New SocketPoll [] owned by 0x7fed08b5f780| net/Socket.cpp:194
coolforkit version details: 21.11.4.2 - 3eff190
frk-00034-00034 2022-05-11 00:12:43.643168 +0000 [ coolforkit ] INF  Have capability cap_sys_chroot| kit/ForKit.cpp:246
frk-00034-00034 2022-05-11 00:12:43.643196 +0000 [ coolforkit ] INF  Have capability cap_mknod| kit/ForKit.cpp:246
frk-00034-00034 2022-05-11 00:12:43.643207 +0000 [ coolforkit ] INF  Have capability cap_fowner| kit/ForKit.cpp:246
frk-00034-00034 2022-05-11 00:12:43.643222 +0000 [ coolforkit ] INF  Have capability cap_chown| kit/ForKit.cpp:246
frk-00034-00034 2022-05-11 00:12:43.643241 +0000 [ coolforkit ] TRC  dlopen(/opt/collaboraoffice/program/libmergedlo.so, RTLD_GLOBAL|RTLD_NOW)| kit/Kit.cpp:2926
frk-00034-00034 2022-05-11 00:12:44.049064 +0000 [ coolforkit ] TRC  Invoking lok_preinit_2(/opt/collaboraoffice/program", "file:///tmp/user")| kit/Kit.cpp:2979

Redis:
1:M 11 May 2022 00:10:51.742 * 1 changes in 3600 seconds. Saving...
1:M 11 May 2022 00:10:51.773 * Background saving started by pid 16
16:C 11 May 2022 00:10:51.811 * DB saved on disk
16:C 11 May 2022 00:10:51.812 * RDB: 0 MB of memory used by copy-on-write
1:M 11 May 2022 00:10:51.874 * Background saving terminated with success

**Reply by szaimen** [reply]
can you send a screenshot of what you see in the aio interface when you enable the firewall?
Does Nextcloud work after you enable it or does it become unavailable directly?

**Reply by hunterwinterton** [reply]
<img width="575" alt="Screen Shot 2022-05-10 at 9 47 18 PM" src="https://user-images.githubusercontent.com/68767432/167765279-54ea313e-c447-4bfe-a2e8-a91e3526d090.png">

They instantly say starting when I turn on the firewall.

**Reply by szaimen** [reply]
I see. Can you post again what you used as startup command for aio?

**Reply by hunterwinterton** [reply]
sudo docker run -it 
--name nextcloud-aio-mastercontainer 
--dns="192.168.2.2" 
--restart always 
-p 8080:8080 
-e APACHE_PORT=11000 
-e NEXTCLOUD_DATADIR="/volume1/docker/nextcloud/data" 
--volume nextcloud_aio_mastercontainer:/mnt/docker-aio-config 
--volume /var/run/docker.sock:/var/run/docker.sock:ro 
nextcloud/all-in-one:beta

**Reply by szaimen** [reply]
okay, that looks fine.

**Reply by szaimen** [reply]
This issue most likely happens because your firewall somehow blocks the connection from the mastercontainer to the other containers in question. So you need to fix your firewall to not do this as it should not be the desired behaviour.

Maybe ask in a synology forum how to do that... enabling ufw on ubuntu does not do this so I think that this is likely something specific to the synology firewall...

**Comment by szaimen** [comment, marked as answer]
This should be fixed with v1.2.0.
